Panic during TestPreparingProgress

## Flaky Test

### Which jobs are failing

github.com/tikv/pd/tests/server/api.TestPreparingProgress

### CI link

https://github.com/tikv/pd/actions/runs/14192003294/job/39758593627

### Reason for failure (if possible)

```bash
--- FAIL: TestPreparingProgress (9.57s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x433144f]

goroutine 236 [running]:
testing.tRunner.func1.2({0x493d240, 0x717ff00})
	/opt/hostedtoolcache/go/1.23.7/x64/src/testing/testing.go:1632 +0x3fc
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.23.7/x64/src/testing/testing.go:1635 +0x6b6
panic({0x493d240?, 0x717ff00?})
	/opt/hostedtoolcache/go/1.23.7/x64/src/runtime/panic.go:791 +0x132
github.com/tikv/pd/tests.MustPutStore(0xc000b1c7c0, 0xc0018ac390, 0xc0004b6780)
	/home/runner/work/pd/pd/tests/testutil.go:217 +0x5af
github.com/tikv/pd/tests/server/api.TestPreparingProgress(0xc0001a2340)
	/home/runner/work/pd/pd/tests/server/api/api_test.go:1073 +0x1b6d
testing.tRunner(0xc0001a2340, 0x5650c98)
	/opt/hostedtoolcache/go/1.23.7/x64/src/testing/testing.go:1690 +0x227
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.23.7/x64/src/testing/testing.go:1743 +0x826
```

--- a/tests/testutil.go
+++ b/tests/testutil.go
@@ -47,7 +47,7 @@ import (
 	"github.com/tikv/pd/pkg/utils/logutil"
 	"github.com/tikv/pd/pkg/utils/testutil"
 	"github.com/tikv/pd/pkg/versioninfo"
-	"github.com/tikv/pd/server"
+	"github.com/tikv/pd/server/cluster"
 )
 
 var (
@@ -193,28 +193,33 @@ func WaitForPrimaryServing(re *require.Assertions, serverMap map[string]bs.Serve
 }
 
 // MustPutStore is used for test purpose.
-func MustPutStore(re *require.Assertions, cluster *TestCluster, store *metapb.Store) {
+func MustPutStore(re *require.Assertions, tc *TestCluster, store *metapb.Store) {
 	store.Address = fmt.Sprintf("tikv%d", store.GetId())
 	if len(store.Version) == 0 {
 		store.Version = versioninfo.MinSupportedVersion(versioninfo.Version2_0).String()
 	}
-	var svr *server.Server
+	var raftCluster *cluster.RaftCluster
 	// Make sure the raft cluster is ready, no matter if the leader is changed.
 	testutil.Eventually(re, func() bool {
-		leader := cluster.GetLeaderServer()
+		leader := tc.GetLeaderServer()
 		if leader == nil {
 			return false
 		}
-		svr = leader.GetServer()
-		return svr != nil
+		svr := leader.GetServer()
+		if svr == nil {
+			return false
+		}
+		raftCluster = svr.GetRaftCluster()
+		// Wait for the raft cluster on the leader to be bootstrapped.
+		return raftCluster != nil && raftCluster.IsRunning()
 	})
-	re.NoError(svr.GetRaftCluster().PutMetaStore(store))
+	re.NoError(raftCluster.PutMetaStore(store))
 	ts := store.GetLastHeartbeat()
 	if ts == 0 {
 		ts = time.Now().UnixNano()
 	}
 
-	storeInfo := svr.GetRaftCluster().GetStore(store.GetId())
+	storeInfo := raftCluster.GetStore(store.GetId())
 	newStore := storeInfo.Clone(
 		core.SetStoreStats(&pdpb.StoreStats{
 			Capacity:  uint64(10 * units.GiB),
@@ -223,9 +228,9 @@ func MustPutStore(re *require.Assertions, cluster *TestCluster, store *metapb.St
 		}),
 		core.SetLastHeartbeatTS(time.Unix(ts/1e9, ts%1e9)),
 	)
-	svr.GetRaftCluster().GetBasicCluster().PutStore(newStore)
-	if cluster.GetSchedulingPrimaryServer() != nil {
-		cluster.GetSchedulingPrimaryServer().GetCluster().PutStore(newStore)
+	raftCluster.GetBasicCluster().PutStore(newStore)
+	if tc.GetSchedulingPrimaryServer() != nil {
+		tc.GetSchedulingPrimaryServer().GetCluster().PutStore(newStore)
 	}
 }
 

